syntax/syntax.go

```diff
@@ -9,6 +9,33 @@ package syntax
 type Node interface {
 	// Span returns the start and end position of the expression.
 	Span() (start, end Position)
+
+	Comment() *Comments
+}
+
+// A Comment represents a single # comment.
+type Comment struct {
+	Start  Position
+	Token  string // without trailing newline
+	Suffix bool   // an end of line (not whole line) comment
```

Comment:
Suffix comments are handled differently from line comments in all places, so rather than put an extra bit (actually a word) in every comment, just keep them separate.  Make the parser have, in place of one field p.comments, two fields lineComments (a slice) and suffixComment (an optional one).

Reply:
As I try to converge this AST with Buildifier's AST, I'd prefer to keep the type like this for now. I'll check what Buildifier exactly does before changing the structure.